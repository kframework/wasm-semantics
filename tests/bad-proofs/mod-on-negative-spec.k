requires "kwasm-lemmas.k"

module MOD-ON-NEGATIVE-SPEC
    imports KWASM-LEMMAS

    rule <k> X modInt N => X ... </k>
      requires X <=Int 0
       andBool 0  <Int N

endmodule
