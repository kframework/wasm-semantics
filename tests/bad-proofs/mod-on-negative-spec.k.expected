  <wasm>
    <k>
      nop ~> X modInt N ~> DotVar2
    </k>
    <valstack>
      _00
    </valstack>
    <curFrame>
      <locals>
        _14
      </locals>
      <localIds>
        _13
      </localIds>
      <curModIdx>
        _12
      </curModIdx>
      <labelDepth>
        _11
      </labelDepth>
      <labelIds>
        _10
      </labelIds>
    </curFrame>
    <moduleRegistry>
      _20
    </moduleRegistry>
    <moduleIds>
      _30
    </moduleIds>
    <moduleInstances>
      _40
    </moduleInstances>
    <nextModuleIdx>
      _50
    </nextModuleIdx>
    <mainStore>
      <funcs>
        _67
      </funcs>
      <nextFuncAddr>
        _66
      </nextFuncAddr>
      <tabs>
        _65
      </tabs>
      <nextTabAddr>
        _64
      </nextTabAddr>
      <mems>
        _63
      </mems>
      <nextMemAddr>
        _62
      </nextMemAddr>
      <globals>
        _61
      </globals>
      <nextGlobAddr>
        _60
      </nextGlobAddr>
    </mainStore>
    <deterministicMemoryGrowth>
      _70
    </deterministicMemoryGrowth>
    <nextFreshId>
      _80
    </nextFreshId>
  </wasm>
#And
  {
    X <=Int 0 andBool 0 <Int N
  #Equals
    true
  }
